ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -2831,3 +2831,84 @@ enum pkcs11_rc add_missing_attribute_id(struct obj_attrs **pub_head,
 	else
 		return set_attribute(pub_head, PKCS11_CKA_ID, id2, id2_size);
 }
+
+enum pkcs11_rc check_attr_checksum(struct obj_attrs *head)
+{
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+
+	uint32_t value_len = 0;
+	uint32_t kcv_len = 0;
+	void *value = NULL;
+	void *kcv = NULL;
+
+	if (!head)
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	/* CKA_CHECK_VALUE is also an attribute of CKO_CERTIFICATE */
+	if (get_class(head) != PKCS11_CKO_SECRET_KEY)
+		return PKCS11_RV_NOT_IMPLEMENTED;
+
+	if (get_key_type(head) != PKCS11_CKK_AES)
```

Comment:
We need to support CKA_CHECK_VALUE for all supported CKO_SECRET_KEY objects, as stated in the spec:
"_The attribute is optional, but if supported, regardless of how the key object is created or derived, the value of the attribute is always supplied._"

`PKCS11_CKK_AES` keys would use `compute_check_value_from_ecb()`.
`PKCS11_CKK_GENERIC_SECRET` would use `compute_check_value_from_sha1()` (compute SHA1 and extract the 3 first bytes).
For `PKCS11_CKK_SHA*_HMAC*`, I understand they fall in the CKK_GENERIC_SECRET case => 3 first byte of the SHA1 hash of the key data.